addons/account/models/account_move.py

```diff
@@ -1210,16 +1210,18 @@ def _compute_tax_totals(self):
                         ))
                 move.tax_totals = self.env['account.tax']._prepare_tax_totals(**kwargs)
                 if move.invoice_cash_rounding_id:
-                    rounding_amount = move.invoice_cash_rounding_id.compute_difference(move.currency_id, move.tax_totals['amount_total'])
+                    rounding_lines = move.line_ids.filtered(lambda line: line.display_type == 'rounding')
+                    rounding_strategy = 'biggest_tax' if rounding_lines.tax_repartition_line_id else 'add_invoice_line'
```

Comment:
Why not use `move.invoice_cash_rounding_id.strategy` like before?

Reply:
because if someone changes in cash rounding method then the old invoice will still show as per the existing calculation not as per the change in cash rounding

Reply:
Then if you change the rounding method (use another one, not modify the existing one), it won't detect and recompute it accordingly either.